src/pytorch_lightning/trainer/trainer.py

```diff
@@ -112,7 +112,8 @@ class Trainer:
     @_defaults_from_env_vars
     def __init__(
         self,
-        logger: Union[Logger, Iterable[Logger], bool] = True,
+        # TODO: Replace default for `None` in 2.0
+        logger: Optional[Union[Logger, Iterable[Logger]]] = _NoneSentinel(),  # type: ignore[assignment]
```

Comment:
Removing `bool` will "break" mypy users still passing a `bool` even in v1.9 (but maybe that's a nice way to signal the deprecation).

Reply:
Exactly. We kind of want to do this as a way to notify the change. Having backwards compatibility for mypy would be too restrictive